anesthetic/plot.py

```diff
@@ -171,6 +176,12 @@ def make_2d_axes(params, **kwargs):
                 sy = sy[0] if sy else None
                 axes[x][y] = fig.add_subplot(grid[j, i],
                                              sharex=sx, sharey=sy)
+
+            if upper == lower and x != y and diagonal:
+                axes[x][y].axis('off')
+                axes[x][y]._upper = None
+                continue
+
```

Comment:
Why did you need to change this implementation? It could be this that is causing the failing tests

Reply:
This is what I was referring to earlier (where I didn't know how to reference the changed lines...).

I personally prefer having all axes created and the unused ones turned invisible. This is useful for later placements of legends or annotations. What are your thoughts?

And yes, this is causing the failing tests. Am happy to adapt the tests for this, but only if you'd be happy with the change.

Reply:
That's a difficult one. I can see pros and cons either way, which means it's probably a good idea to make this some kind of optional behaviour (e.g. with an 'invisible' argument). Revert for now, as it's definitely a separate change with regards to this PR, but submit this as an issue for further discussion (we can also use that to discuss how to make ``make_2d_axes`` more consistent with the new ``'upper', 'lower', 'diagonal'`` scheme.

Reply:
reverted.
tests now passing :)